pkg/scheduler/plugins/elasticquota/core/group_quota_manager.go

```diff
@@ -0,0 +1,616 @@
+					newSubGroupsTotalRes, subQuotaInfo.ParentName, subQuotaInfo.Name)
+				if needScale {
+					gqm.doUpdateOneGroupAutoScaleMinQuotaNoLock(subQuotaInfo, newMinQuota)
+				}
+			}
+
+			//3. update sub-group's runtimeQuota
+			if subQuotaInfo.RuntimeVersion != subTreeWrapper.GetVersion() {
+				subTreeWrapper.UpdateOneGroupRuntimeQuota(subQuotaInfo)
+			}
+		}
+
+		return curToAllParInfos[0].getMaskedRuntimeNoLock()
+	}
+}
+
+// doUpdateOneGroupAutoScaleMinQuotaNoLock no need to lock gqm.lock
+func (gqm *GroupQuotaManager) doUpdateOneGroupAutoScaleMinQuotaNoLock(quotaInfo *QuotaInfo, newMinRes v1.ResourceList) {
+	if !quotav1.Equals(quotaInfo.CalculateInfo.AutoScaleMin, newMinRes) {
+		quotaInfo.setAutoScaleMinQuotaNoLock(newMinRes)
+		gqm.quotaTreeWrapperMap[quotaInfo.ParentName].UpdateOneGroupMinQuota(quotaInfo)
+	}
+}
+
+func (gqm *GroupQuotaManager) getCurToAllParentGroupQuotaInfoNoLock(quotaName string) []*QuotaInfo {
+	curToAllParInfos := make([]*QuotaInfo, 0)
+	quotaInfo := gqm.getGroupQuotaInfoByNameNoLock(quotaName)
+	if quotaInfo == nil {
+		return curToAllParInfos
+	}
+
+	for true {
+		curToAllParInfos = append(curToAllParInfos, quotaInfo)
+		if quotaInfo.ParentName == extension.RootQuotaName || quotaInfo.ParentName == extension.SystemQuotaName ||
+			quotaInfo.ParentName == extension.DefaultQuotaName {
+			break
+		}
+
+		quotaInfo = gqm.getGroupQuotaInfoByNameNoLock(quotaInfo.ParentName)
+		if quotaInfo == nil {
+			return curToAllParInfos
+		}
+	}
+
+	return curToAllParInfos
+}
+
+func (gqm *GroupQuotaManager) GetGroupQuotaInfoByName(quotaName string) *QuotaInfo {
+	gqm.Lock.RLock()
+	defer gqm.Lock.RUnlock()
+
+	return gqm.getGroupQuotaInfoByNameNoLock(quotaName)
+}
+
+func (gqm *GroupQuotaManager) getGroupQuotaInfoByNameNoLock(quotaName string) *QuotaInfo {
+	return gqm.quotaInfoMap[quotaName]
+}
+
+func (gqm *GroupQuotaManager) getTreeWrapperByNameNoLock(quotaName string) (treeWrapperPtr *QuotaTreeWrapper) {
+	return gqm.quotaTreeWrapperMap[quotaName]
+}
+
+func (gqm *GroupQuotaManager) scopedLockForQuotaInfo(quotaList []*QuotaInfo) func() {
+	listLen := len(quotaList)
+	for i := listLen - 1; i >= 0; i-- {
+		quotaList[i].lock.Lock()
+	}
+
+	return func() {
+		for i := 0; i < listLen; i++ {
+			quotaList[i].lock.Unlock()
+		}
+	}
+}
+
+func (gqm *GroupQuotaManager) DealWithQuota(quota *v1alpha1.ElasticQuota, isDelete bool) error {
+	gqm.Lock.Lock()
+	defer gqm.Lock.Unlock()
+
+	quotaName := quota.Name
+	inputQuotaGroups := make([]*QuotaInfo, 0)
+	if isDelete {
+		_, exist := gqm.quotaInfoMap[quotaName]
+		if !exist {
+			return fmt.Errorf("get quota info failed, quotaName:%v", quotaName)
+		}
+		delete(gqm.quotaInfoMap, quotaName)
+	} else {
+		quotaInfo := NewQuotaInfoFromQuota(quota)
+		// if update/create DefaultQuotaGroup, no need to resetTree.
+		if quotaName == extension.DefaultQuotaName {
+			quotaInfo.CalculateInfo.Used = gqm.quotaInfoMap[extension.DefaultQuotaName].GetUsed()
```

Comment:
why you update quotaInfo.CalculateInfo's used\request\runtime by crd not update updateLogic when receive pod!???